src/modal/testing/modal.service.spec.ts

```diff
@@ -37,47 +37,69 @@ describe('Modal service', () => {
     fixture.detectChanges();
   });
 
-  it('should return random id on spin up a new modal', done => {
+  it('should return random id on spin up a new modal', () => {
     modalService.onShown.subscribe((data) => {
       expect(data.id).toBeTruthy();
-      done();
     });
 
     modalService.show(TestModalComponent);
   });
 
-  it('should return different random id when open nested modal without specifying id', done => {
+  it('should return different random id when open nested modal without specifying id', () => {
     modalService.onShown.pipe(
       pairwise(),
       tap(([firstData, secondData]) => {
         expect(firstData.id).not.toBe(secondData.id);
-        done();
       })
     ).subscribe();
 
     modalService.show(TestModalComponent);
     modalService.show(TestModalComponent);
   });
 
-  it('should return id in config when specified', done => {
+  it('should return id in config when specified', () => {
     const id = 20;
 
     modalService.onShown.subscribe((data) => {
       expect(data.id).toBe(id);
-      done();
     });
 
     modalService.show(TestModalComponent, { id });
   });
 
   it('should return id when hide modal', () => {
     const id = 20;
+    jest.useFakeTimers();
 
     modalService.onHidden.subscribe((data) => {
       expect(data.id).toBe(id);
     });
 
     const bsRef = modalService.show(TestModalComponent, { id });
     bsRef.hide();
+
+    jest.runAllTimers();
+  });
+
+  it('should hide modal even if invoked right after show', () => {
+    const id = 25;
+    const onHideSpy = jest.spyOn(modalService.onHide, 'emit');
+    const onHiddenSpy = jest.spyOn(modalService.onHidden, 'emit');
+    jest.useFakeTimers();
+
+    modalService.onHidden.subscribe((data) => {
```

Comment:
I see that this test is taken by the test `should return id when hide modal`.

I would suggest skipping that test and focusing on testing one thing.
If it fails you would isolate what the problem is, otherwise the problem can be in several places.

I found [this](https://stackoverflow.com/questions/235025/why-should-unit-tests-test-only-one-thing/235032#235032) on the internet and explain better my point

Reply:
Got it, done.